semgrep-core/parsing/Parse_kotlin_tree_sitter.ml

```diff
@@ -1404,7 +1413,7 @@ and primary_expression (env : env) (x : CST.primary_expression) : AST.expr =
       in
       let v4 = token env v4 (* "]" *) in
       todo env (v1, v2, v3, v4)
-  | `This_exp tok -> let _ = token env tok in raise Todo (* "this" *)
+  | `This_exp tok -> Id(str env tok, empty_id_info()) (* "this" *)
```

Comment:
There is a IdSpecial for This and Super.
See what I do in Parse_java_tree_sitter.ml